railgun/internal/ctrlutils/utils.go

```diff
@@ -141,3 +146,31 @@ func CRDExists(client client.Client, gvr schema.GroupVersionResource) bool {
 	}
 	return true
 }
+
+// retrieve Kong Admin API URL from configured name/namespace service
+func RetrieveKongAdminAPIURL(ctx context.Context, KongAdminAPI string, kubeCfg *rest.Config) (string, error) {
+	namespace, name, err := util.ParseNameNS(KongAdminAPI)
+	if err != nil {
+		return "", fmt.Errorf("failed to pars kong admin api namespace and name. err %v", err)
+	}
+	CoreClient, _ := clientset.NewForConfig(kubeCfg)
+	svc, err := CoreClient.CoreV1().Services(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to creating kubernetes client")
```

Comment:
This error message is not accurate, because the error is coming from the `Services().Get`, not the clienset creation. Also we can wrap the encountered error for more logging visibility:
```suggestion
		return "", fmt.Errorf("failed retrieve service object %s/%s: %w", namespace, name, err)
```